### PR TITLE
Add min max validator

### PR DESCRIPTION
**Pull request by Dufgui**
fix #7514

### CODE REVIEW

**Comment by mraible** [comment]
If it's not too much trouble, I'd love to see a test or two.

**Comment by Dufgui** [comment]
tu in progress ;o)

**Comment by mraible** [comment]
Great! Thanks and congrats on your first contribution. 🎉

**Comment by Dufgui** [comment]
tks, yes it's my first on this side. ;o)

File: tests/directive/max.spec.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/*
+ Copyright 2013-Present the original author or authors from the JHipster project.
```
Comment (deepu105) [inline tests/directive/max.spec.ts]: @Dufgui can you please rename the file to `max.directive.spec.ts` same for other file

**Comment by deepu105** [comment]
@Dufgui this has been released. Could you update the generator use this?

**Comment by Dufgui** [comment]
I will take a look at it this week. Thanks
Le ven. 4 mai 2018 à 10:26, Deepu K Sasidharan <notifications@github.com> a
écrit :

File: src/directive/min.directive.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        { provide: NG_VALIDATORS, useExisting: forwardRef(() => JhiMinValidatorDirective), multi: true }
+    ]
+})
+export class JhiMinValidatorDirective {
+    @Input() jhiMin: number;
+
+    constructor() {}
+
+    validate(c: FormControl) {
+        return (c.value && c.value >= this.jhiMin) ? null : {
```
[the remark this answers is not in the document]
Comment (ruddell) [reply src/directive/min.directive.ts]: 0 > "0" === false

There's a spec written that tests your example https://github.com/jhipster/ng-jhipster/blob/master/tests/directive/min.directive.spec.ts#L38-L42

Are you sure you have the latest ng-jhipster installed in your project?

[the remark this answers is not in the document]
Comment (ruddell) [reply src/directive/min.directive.ts]: Oh I see what you mean by 0 being false, but it seems to work in a generated project

**Pull request merged**